shark-cli/src/main/java/shark/Main.kt

```diff
@@ -124,7 +124,7 @@ private fun runCommand(
   directory: File,
   vararg arguments: String
 ): String {
-  val process = ProcessBuilder(*arguments)
```

Comment:
what's wrong with the spread operator btw?

Reply:
From the [rule description:](https://arturbosch.github.io/detekt/performance.html#spreadoperator)

> SpreadOperator
In most cases using a spread operator causes a full copy of the array to be created before calling a method. This has a very high performance penalty. Benchmarks showing this performance penalty can be seen here: https://sites.google.com/a/athaydes.com/renato-athaydes/posts/kotlinshiddencosts-benchmarks
> The Kotlin compiler since v1.1.60 has an optimization that skips the array copy when an array constructor function is used to create the arguments that are passed to the vararg parameter. When type resolution is enabled in detekt this case will not be flagged by the rule since it doesn’t suffer the performance penalty of an array copy.

In this particular case, it doesn't really matter. Using spread operator will result in invoking this constructor, that loops over arguments and adds commands one by one:

```
public ProcessBuilder(String... command) {
    this.command = new ArrayList<>(command.length);
    for (String arg : command)
        this.command.add(arg);
}
```
While converting them to lo list via `.asList()` will first use `ArraysUtilJVM.asList()` to convert array to list and then it will call constructor which will use the list:
```
public ProcessBuilder(List<String> command) {
    if (command == null)
        throw new NullPointerException();
    this.command = command;
}
```